From ad5082845cdde44bbee9ca9d765cdd3b14619e69 Mon Sep 17 00:00:00 2001
From: Kenny Levinsen <kl@kl.wtf>
Date: Sun, 21 Apr 2024 16:42:08 +0200
Subject: [PATCH 14/15] server: Use wlr_renderer_get_texture_formats

wlr_renderer_get_{dmabuf|shm}_texture_formats have been replaced by a
unified wlr_renderer_get_texture_formats interface using buffer caps.

References: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4644
Origin: Upstream
Upstream-Status: Backported [commit ffcde7a70c1341d1ec1d38c00ff87faa1a816892]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 sway/server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sway/server.c b/sway/server.c
index 81df98a07727..cc032148c99c 100644
--- a/sway/server.c
+++ b/sway/server.c
@@ -146,7 +146,7 @@ bool server_init(struct sway_server *server) {
 
 	wlr_renderer_init_wl_shm(server->renderer, server->wl_display);
 
-	if (wlr_renderer_get_dmabuf_texture_formats(server->renderer) != NULL) {
+	if (wlr_renderer_get_texture_formats(server->renderer, WLR_BUFFER_CAP_DMABUF) != NULL) {
 		server->linux_dmabuf_v1 = wlr_linux_dmabuf_v1_create_with_renderer(
 			server->wl_display, 4, server->renderer);
 	}
-- 
2.46.0

